api/devices/schema.py

```diff
@@ -0,0 +1,67 @@
+import graphene
+from datetime import datetime
+from graphene_sqlalchemy import SQLAlchemyObjectType
+
+from api.devices.models import Devices as DevicesModel
+from utilities.utility import validate_empty_fields, update_entity_fields
+
+
+class Devices(SQLAlchemyObjectType):
+    class Meta:
+        model = DevicesModel
+
+
+class CreateDevice(graphene.Mutation):
+    class Arguments:
+        name = graphene.String(required=True)
+        device_type = graphene.String(required=True)
+        location = graphene.String()
+        resource_id = graphene.Int(required=True)
+    device = graphene.Field(Devices)
+
+    def mutate(self, info, **kwargs):
+        device = DevicesModel(
+            **kwargs,
+            date_added=datetime.now(),
+            last_seen=datetime.now()
+        )
+        device.save()
+
+        return CreateDevice(device=device)
+
+
+class UpdateDevice(graphene.Mutation):
+    class Arguments:
+        device_id = graphene.Int()
+        name = graphene.String()
+        device_type = graphene.String()
+        location = graphene.String()
+        resource_id = graphene.Int()
+    device = graphene.Field(Devices)
+
+    def mutate(self, info, device_id, **kwargs):
+        validate_empty_fields(**kwargs)
+
+        query_device = Devices.get_query(info)
+        exact_device = query_device.filter(
+            DevicesModel.id == device_id
+            ).first()
+        exact_device.date_added = datetime.now()
```

Comment:
Before executing any logic with the device object, first check that it actually exists?